src/mame/drivers/megadriv_rad.cpp

```diff
@@ -300,6 +515,32 @@ ROM_START( dgunl3227 )
 ROM_END
 
 
+
+ROM_START( ra145 )
+	/*
+	Data for the following games is corrupt (ranges approximate, based on areas of inconsistent readout)
+
+	3c0000 - 43ffff - James Bond 007 - The Duel   (400171 - 41c600 is corrupt)
+	4c0000 - 53ffff - Robocop 3                   (507bd4 - 50ad43 is corrupt)
+	540000 - 5bffff - Hokuto no Ken               (540006 - 55fff2 is corrupt)
+	640000 - 6bffff - Batman                      (640042 - 65ffaf is corrupt)
+	7c0000 - 7fffff - Rambo 3                     (7e031a - 7fffb9 is corrupt)
+
+	Unfortunately as many of the games in this unit have been hacked, or are using pirate versions of the games
+	from the mid 90s (in a few cases, complete with trainers) that seem to have dropped out of circulation it
+	is not possible to repair the data in dump from the damaged unit.
+
+	The unit also includes a duplicate copy of Thunder Force II instead of Revenge of Shinobi, this however
+	is not a dump issue, nor is Wacky Worlds being swapped with Fun and Games in the menu
+	*/	
+
+	ROM_REGION( 0x8000000, "maincpu", ROMREGION_ERASE00 )
+	ROM_LOAD16_WORD_SWAP( "ra145.bin", 0x000000, 0x8000000, CRC(30583950) SHA1(855eae232e3830a505f9bc1a26edb3a7d15ce4d1) )
```

Comment:
If the dump is corrupt beyond repair, shouldn't it have a BAD_DUMP flag?

Reply:
yes sorry, thought I'd put one there